src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -2980,14 +3009,22 @@ MultiRouterPlannableQuery(Query *query)
 			if (!IsDistributedTable(distributedTableId))
 			{
 				/* local tables cannot be read from workers */
-				return false;
+				distributedPlan->planningError = DeferredError(
+					ERRCODE_FEATURE_NOT_SUPPORTED,
+					"Local tables cannot be used in distributed queries.",
+					NULL, NULL);
+				return;
 			}
 
 			partitionMethod = PartitionMethod(distributedTableId);
 			if (!(partitionMethod == DISTRIBUTE_BY_HASH || partitionMethod ==
 				  DISTRIBUTE_BY_NONE || partitionMethod == DISTRIBUTE_BY_RANGE))
 			{
-				return false;
+				distributedPlan->planningError = DeferredError(
+					ERRCODE_FEATURE_NOT_SUPPORTED,
+					"Unsupported partition method.",
```

Comment:
Maybe: "Router planner does not support append-partitioned tables" or "Router planner only supports hash- and range-distributed tables and reference tables"